specs/simpleserialize.md

```diff
@@ -0,0 +1,230 @@
+ + [Serialize/Encode](#serializeencode)
+   - [uint: 8/16/24/32/64/256](#uint-816243264256)
+   - [Address](#address)
+   - [Hash32](#hash32)
+   - [Bytes](#bytes)
+   - [List](#list)
+ + [Deserialize/Decode](#deserializedecode)
+   - [uint: 8/16/24/32/64/256](#uint-816243264256-1)
+   - [Address](#address-1)
+   - [Hash32](#hash32-1)
+   - [Bytes](#bytes-1)
+   - [List](#list-1)
+* [Implementations](#implementations)
+
+## About
+
+`SimpleSerialize` was first proposed by Vitalik Buterin as the serialization
+protocol for use in the Ethereum 2.0 Beacon Chain.
+
+The core feature of `ssz` is the simplicity of the serialization with low
+overhead.
+
+## Terminology
+
+| Term         | Definition                                                                                     |
+|:-------------|:-----------------------------------------------------------------------------------------------|
+| `big`        | Big Endian                                                                                     |
+| `byte_order` | Specifies [endianness:](https://en.wikipedia.org/wiki/Endianness) Big Endian or Little Endian. |
+| `len`        | Length/Number of Bytes.                                                                        |
+| `to_bytes`   | Convert to bytes. Should take parameters ``size`` and ``byte_order``.                          |
+| `from_bytes` | Convert form bytes to object. Should take ``bytes`` and ``byte_order``.                        |
+| `value`      | The value to serialize.                                                                        |
+| `rawbytes`   | Raw serialized bytes.                                                                          |
+
+## Constants
+
+| Constant       | Value | Definition                                                              |
+|:---------------|:-----:|:------------------------------------------------------------------------|
+| `LENGTH_BYTES` |   4   | Number of bytes used for the length added before the serialized object. |
+
+
+## Overview
+
+### Serialize/Encode
+
+#### uint: 8/16/24/32/64/256
+
+Convert directly to bytes the size of the int. (e.g. ``uint16 = 2 bytes``)
+
+All integers are serialized as **big endian**.
+
+| Check to perform                 | Code                    |
+|:---------------------------------|:------------------------|
+| Size is a byte integer           | ``int_size % 8 == 0``   |
+| Value is less than max           | ``2**int_size > value`` |
+
+```python
+buffer_size = int_size / 8
+return value.to_bytes(buffer_size, 'big')
+```
+
+#### Address
+
+The address should already come as a hash/byte format. Ensure that length is
+**20**.
+
+| Check to perform       | Code                 |
+|:-----------------------|:---------------------|
+| Length is correct (20) | ``len(value) == 20`` |
+
+```python
+assert( len(value) == 20 )
+return value
+```
+
+#### Hash32
+
+The hash32 should already be a 32 byte length serialized data format. The safety
```

Comment:
hash32 works for Keccak/Blake2b[0 ..< 32] not for BLS

Reply:
Yep, great point! Did we address in point above?
 
I think we should have the `Hash97` and `Hash96` separately or address similar to the way `uint` is serialized/deserialized?
Thoughts?

Reply:
I think like `uint`